### ISSUE TITLE
Release 1.1.1 fails (unless used on root context).

### ISSUE THREAD

**Issue by Dehumanizer77** [igniterealtime/openfire-httpFileUpload-plugin#12]
I have just upgraded httpFileUpload plugin to version 1.1.1 and it has stopped working. When trying to send an image, I'm just getting "delivery failed" every time. After switching back to 1.1.0, everything works again.
Client I was testing it with is Conversations version 2.5.12+pcr (on Android 9)

**Comment by wrooot** [comment]
Server version?

**Comment by fuxuan** [comment]
same Openfire 4.4.3
the PUT is https://xxx.com/{guid}/xxx.zip, no /httpfileupload in it.

**Comment by wrooot** [comment]
Ugh, Spark stops working every time plugin updates. So i don't have a mean to test this locally (can't connect with Conversations to my local server, Gajim also doesn't want to send file because of a self-signed cert).

@guusdk @akrherz  is this plugin updated on ignite server? Then we might be able to test it with Conv/Gajim.

**Comment by akrherz** [comment]
> @akrherz is this plugin updated on ignite server?

I have updated it now.

**Comment by wrooot** [comment]
Failing to upload in both xabber and conv

**Comment by fenuz** [comment]
+1

It seems that the advertised URL is not the url the webapp is mounted at (`/httpfileupload`). The advertised URL's are in the `uuid/filename.extension` format. The old httpfileuploadcomponent (1.1.3) had an `endpoint` set to `Component`, which was used as the slot URL prefix, the new (1.2.0) version does not have such a construct.

I am under the impression that the 1.1.1 version of this plugin does not work at all, or am I missing a workaround? Perhaps its best to remove the plugin download from the openfire plugin downloads while this gets fixed.

**Comment by wrooot** [comment]
It looks like it doesn't work indeed.

**Comment by guusdk** [comment]
Yeah, my bad. Release 1.1.1 appears to functioning only if it's being ran through a reverse proxy of sorts, mapping the endpoint context to a root context of a domain. I'll resolve this shortly.

**Comment by Dehumanizer77** [comment]
Still not working. Plus the server behaves very strangely when I install 1.1.1-SNAPSHOT. Messages not getting sent/received randomly, I had to remove the plugin completely and also restart openfire completely to put it back to normal. Then installed 1.1.0 back and again everything is working. I have tried it several times. 
Server version 4.4.3
Sorry for not responding earlier but for some reason I didn't get an email notification from github until the issue had been closed.
